Assets/Scripts/UI/AchievementPatches.cs

```diff
@@ -0,0 +1,121 @@
+            numDeaths, Sprite levelCandy)
+    {
+        //conditions to meet to get patch
+        bool candyCondition = candiesInLevel - candiesCollected == 0;
+        bool deathCondition = numDeaths == 0;
+        bool timeCondition = WithinTimeThreshold(level, timeTaken);
+        
+        //sets the candy image (because it changes depending on level)
+        defaultCandy = levelCandy;
+        
+        //check default value for candy
+        if (!candiesInLevel.Equals(-1))
+            candyImage.sprite = candyCondition ? candyPatch : defaultCandy;
+        
+        deathImage.sprite = deathCondition? deathPatch:defaultDeath;
+        
+        //check default value for time
+        if (!float.IsNaN(timeTaken))
+            timeImage.sprite = timeCondition ? timePatch : defaultTime;
+        else
+            timeImage.sprite = defaultTime;
+        
+        //apply gold to playPanel if applicable
+        if (timeCondition && candyCondition && deathCondition)
+            playPanel.sprite = goldPanel;
+        else 
+            playPanel.sprite = defaultPanel;
+    }
+
+    private bool WithinTimeThreshold(int level, float timeTaken)
+    {
+        return level switch
+        {
+            1 => timeTaken < level1Threshold,
+            2 => timeTaken < level2Threshold,
+            3 => timeTaken < level3Threshold,
+            4 => timeTaken < level4Threshold,
+            _ => false
+        };
+    }
+
+    /// <summary>
+    /// Determines whether we meet the 3 conditions of setting the button gold:
+    /// 1. Under time threshold
+    /// 2. All Candies Collected
+    /// 3. Zero deaths
+    /// called from LevelSelect line 46
+    /// </summary>
+    /// <param name="btn">Candidate button for turning gold</param>
+    /// <param name="level">Level number</param>
+    public void CheckSetGoldButton(LevelSelectButton btn, int level)
+    {
+        GameManager gm = GameManager.Instance;
+        LevelData thisLevel = gm.AllLevelData[level - 1];
+
+        //check defaults, then check if we meet all requirement
+        if (!thisLevel.totalCandiesInLevel.Equals(-1) &&
+            !thisLevel.leastDeaths.Equals(-1) &&
+            !float.IsNaN(thisLevel.bestTime))
+        {
+            if (WithinTimeThreshold(level, thisLevel.bestTime) &&
+                thisLevel.mostCandiesCollected - thisLevel.totalCandiesInLevel == 0 &&
+                thisLevel.leastDeaths == 0)
+            {
+                btn.SetGold();
+                SetVariableGold(level);
+            }
+        }
+    }
+
+    private void SetVariableGold(int level)
+    {
+        switch (level)
+        {
+            case 1:
+                isGoldL1 = true;
+                break;
+            case 2:
+                isGoldL2 = true;
+                break;
+            case 3:
+                isGoldL3 = true;
+                break;
+            case 4:
+                isGoldL4 = true;
+                break;
+        }
+    }
```

Comment:
tbh kinda hate having to do this kinda stuff everywhere but hey we're having 4 fixed levels so ig we're doing that now